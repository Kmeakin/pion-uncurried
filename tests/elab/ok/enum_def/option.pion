enum Option(T: Type) {
    None,
    Some(x: T),
}

let n = None;
let s = Some;

let unwrap_or: fn(_: Option(Bool), _: Bool) -> Bool 
= fn(o: Option(Bool), d: Bool) => 
    match o {
        Some(y) => y, 
        None => d, 
    };
