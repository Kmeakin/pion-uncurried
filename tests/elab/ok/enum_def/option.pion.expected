enum Option(T: Type): Type {
    None: Option(T),
    Some(x: T): Option(T),
}

let n: fn(T: Type) -> Option(T) = None;

let s: fn(T: Type, x: T) -> Option(T) = Some;
